demos/segmentation_demo/main.cpp

```diff
@@ -89,178 +64,141 @@ int main(int argc, char *argv[]) {
             slog::info << "GPU Extension loaded: " << FLAGS_c << slog::endl;
         }
 
-        /** Printing device version **/
-
         slog::info << "Device info" << slog::endl;
         std::cout << ie.GetVersions(FLAGS_d);
 
-        // -----------------------------------------------------------------------------------------------------
-
-        // --------------------------- 2. Read IR Generated by ModelOptimizer (.xml and .bin files) ------------
```

Comment:
Why did you remove these headings?

Reply:
Demos are not IE API examples. If one doesn't get what is happening at that level (like read net, load net, create InferRequest), one should refer to samples. Though it can be useful to have such comments explaining pre/prost processing steps and overall processing pipeline (from reading an image to showing results) solutions, such comments shouldn't focus on API.

Reply:
> Demos are not IE API examples. If one doesn't get what is happening at that level (like read net, load net, create InferRequest), one should refer to samples.

Fair point.

> Though it can be useful to have such comments explaining pre/prost processing steps and overall processing pipeline (from reading an image to showing results) solutions, such comments shouldn't focus on API.

Sure, but you didn't add such comments. :-)

I'm not saying that these headings should stay as-is, I just think they're useful in principle to break up the function into smaller chunks.

Reply:
> Sure, but you didn't add such comments. :-)

That's why I said can, but not must :)

> they're useful in principle to break up the function into smaller chunks.

I split blocks of code with a single line.